request/http.go

```diff
@@ -0,0 +1,70 @@
+package request
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	// HTTPHeaderID is the HTTP Header to be used when sending ID in
+	// HTTP Requests or Responses
+	HTTPHeaderID = "X-REQUESTID"
+)
+
+// GetFromHTTPRequest attempts to read a Request ID from the given @r http request's
+// header. If no ID is found, or the found ID is ill-formatted, an empty ID is returned.
+func GetFromHTTPRequest(r *http.Request) ID {
+	idStr := r.Header.Get(HTTPHeaderID)
+
+	id, err := Parse(idStr)
+	if err != nil {
+		return ID{}
+	}
+
+	return id
+}
+
+// SetInHTTPRequest will add the ID registered in @ctx into the given
+// @request as a HTTP Header. If @request is nil, an warning log will
+// be emitted and nothing will be changed.
+func SetInHTTPRequest(ctx context.Context, request *http.Request) {
+	if request == nil {
+		log.Warn().
+			Str("method", "request.SetInHTTPRequest").
+			Msg("[FoundationKit] Request is nil. Nothing will happen")
+		return
+	}
+
+	id := GetRequestIDFromContext(ctx)
+	request.Header.Set(HTTPHeaderID, id.String())
+}
+
+// GetFromHTTPResponse attempts to read a Request ID from the given @r http response's
+// header. If no ID is found, or the found ID is ill-formatted, an empty ID is returned.
+func GetFromHTTPResponse(r *http.Response) ID {
+	idStr := r.Header.Get(HTTPHeaderID)
+
+	id, err := Parse(idStr)
+	if err != nil {
+		return ID{}
+	}
+
+	return id
+}
+
+// SetInHTTPResponse will add the ID registered in @ctx into the given
+// @request as a HTTP Header. If @request is nil, an warning log will
+// be emitted and nothing will be changed.
+//
+// nolint: interfacer
+func SetInHTTPResponse(id ID, response http.ResponseWriter) {
+	if response == nil {
+		log.Warn().
```

Comment:
I think this should panic because it's the purpose of the function to set the id on the response writer.  So no need for this check.

Reply:
The `trace` package defines the methods by using warnings instead of `panic`. Since these methods are already "widely" used, I think it's unhealthy to change their contract behavior. Then, it would also be unhealthy that `trace` packages uses warnings, but the `request` equivalents panics on sanity failures.

My opinion is that keeping the same behavior as `trace` is a stronger factor here, and if we want to change the behavior from warning to panic in both packages, it would be better to have multiple versions of the method (one for warning, and one for panic) to keep the contract "error" behavior stable.